### PR TITLE
DBZ-5064 Use Maven 3.8.4 for GitHub CI

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-5064

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -316,7 +316,7 @@
                         <configuration>
                             <rules>
                                 <requireMavenVersion>
-                                    <version>3.6.3</version>
+                                    <version>3.8.4</version>
```
Comment (vjuranek) [inline pom.xml:319]: Looks like this enforce is not used. I'd either use (preferred) it or remove it.

**Review by vjuranek — COMMENTED** [review]
I think right now we use default (i.e. random) maven version in github workflows. IMHO we should add some [maven action](https://github.com/marketplace?type=actions&query=maven) (e.g. [sdkman](https://github.com/sdkman/sdkman-action)) to use maven with specified version.

File: pom.xml (hunk 1 above)
Comment (jpechane) [reply pom.xml]: @vjuranek Please create a Jira for enforcing

**Comment by jpechane** [comment]
@ani-sha Thanks

File: pom.xml (hunk 1 above)
Comment (vjuranek) [reply pom.xml]: https://issues.redhat.com/browse/DBZ-5069

**Pull request merged**